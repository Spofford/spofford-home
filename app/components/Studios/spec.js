import React from "react"
import expect from "expect"
import { shallow } from "enzyme"

import { Studios } from "./"

describe("<Studios />", () => {
  it("should render", () => {
    const renderedComponent = shallow(
      <Studios />
    )
    expect(renderedComponent.is("div")).toEqual(true)
  })
})
