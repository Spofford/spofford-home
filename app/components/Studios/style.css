.studios {

}

.studios img {
  width:100%;
}

.studios .copy-container h2 {

}

.studios .copy-container p {
  line-height:1.8em;
  margin-bottom:40px;
  font-family:'Montserrat', sans-serif;
  font-size:1.5em;
}

.studios .fullHeader .container {
  background-color:#33CB2B;
  box-shadow:none;
}

.studios .hero h3, .studios .hero p, .studios > .fullHeader h2, .studios > .fullHeader > .container .fa-bars, .studios > .fullHeader > .container .menu-label, .studios > .fullHeader .search-container.hidden .fa-search {
  color:white;
}

.studios .hero {
  background-color: #33CB2B;
  height:calc(100% - 65px);
  margin-top:65px;
}

.studios .copy-container hr {
  margin: 60px 0px 0px;
}

.studios .fa-chevron-right {
  margin-left:10px;
}

@media (min-width: 700px) {
	.studios .hero .copy-container {
    max-width:61.8%;
    padding-left:60px;
  }
}
