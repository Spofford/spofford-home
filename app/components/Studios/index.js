import React from "react"
import * as contentful from 'contentful'
const ReactMarkdown = require('react-markdown')
import style from "./style.css"
import { Link } from 'react-router-dom'
import FontAwesome from "react-fontawesome";

import { default as Header } from "../Header"

export class Studios extends React.Component {
  state = {
    model: {},
    asset1: "",
    asset2: "",
    asset3: ""
  }

  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })

  componentDidMount() {
    /* window.scrollTo(0, 0); */
    this.fetchModel().then(this.setModel);
  }

  fetchModel = () => this.client.getEntry('68s1X4SuTmk6CS8oc2couM')

  setModel = response => {
    var self = this
    console.log(response)
    this.setState({
      model: response.fields
    })
    self.client.getAsset(this.state.model.visual.sys.id).then(self.setAsset1)
    self.client.getAsset(this.state.model.visual2.sys.id).then(self.setAsset2)
    self.client.getAsset(this.state.model.visual3.sys.id).then(self.setAsset3)
  }

  setAsset1 = response => {
    this.setState({
      asset1: response.fields.file.url
    })
  }

  setAsset2 = response => {
    this.setState({
      asset2: response.fields.file.url
    })
  }

  setAsset3 = response => {
    this.setState({
      asset3: response.fields.file.url
    })
  }



  render() {
    return (
      <div className="studios">
        <Header />
        <div className="hero">
          <div className="copy-container">
            <h2>{this.state.model.head}</h2>
            <ReactMarkdown source={this.state.model.bodyText} />
          </div>
        </div>
        <div className="body-container">

        <div className="copy-container">
          <h2>{this.state.model.subhead4}</h2>
          <img src={this.state.asset3} />
          <ReactMarkdown source={this.state.model.bodyText4} />
          <Link to='/show'>Submit a design<FontAwesome name='chevron-right' /></Link>
        </div>

          <div className="copy-container">
            <h2>{this.state.model.subhead2}</h2>
            <img src={this.state.asset1} />
            <ReactMarkdown source={this.state.model.bodyText2} />
            <Link to='/feedback/start'>view designs<FontAwesome name='chevron-right' /></Link>
            <hr />
          </div>

          <div className="copy-container">
            <h2>{this.state.model.subhead3}</h2>
            <img src={this.state.asset2} />
            <ReactMarkdown source={this.state.model.bodyText3} />
            <a target="_blank" href="http://may-insights.spofforddesign.com/">read on<FontAwesome name='chevron-right' /></a>
          </div>
        </div>
      </div>
    )
  }
}

export default Studios
