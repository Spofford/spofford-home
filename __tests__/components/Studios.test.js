import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Studios from '../../app/components/Studios';

describe('<Studios />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Studios />);
      // const component = wrapper.dive();

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('calls setModel on mount', async () => {
      const wrapper = shallow(<Studios />);
      const spy = jest.spyOn(wrapper.instance(), "setModel");
      wrapper.instance().forceUpdate();
      wrapper.instance().componentDidMount();

      const data = await wrapper.instance().fetchModel();
      expect(spy).toHaveBeenCalled();

      spy.mockRestore();
    })
  });
});
