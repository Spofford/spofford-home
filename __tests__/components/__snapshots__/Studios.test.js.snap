// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Studios /> render() renders the component 1`] = `
<div
  className="studios"
>
  <Connect(Header) />
  <div
    className="hero"
  >
    <div
      className="copy-container"
    >
      <h2 />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
    </div>
  </div>
  <div
    className="body-container"
  >
    <div
      className="copy-container"
    >
      <h2 />
      <img
        src=""
      />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        replace={false}
        to="/show"
      >
        Submit a design
        <FontAwesome
          name="chevron-right"
        />
      </Link>
    </div>
    <div
      className="copy-container"
    >
      <h2 />
      <img
        src=""
      />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        replace={false}
        to="/feedback/start"
      >
        view designs
        <FontAwesome
          name="chevron-right"
        />
      </Link>
      <hr />
    </div>
    <div
      className="copy-container"
    >
      <h2 />
      <img
        src=""
      />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <a
        href="http://may-insights.spofforddesign.com/"
        target="_blank"
      >
        read on
        <FontAwesome
          name="chevron-right"
        />
      </a>
    </div>
  </div>
</div>
`;
